hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -832,12 +831,18 @@ private SortOrder interpretSortOrder(String value) {
 		return entityReference;
 	}
 
-	private EntityDomainType resolveEntityReference(String entityName) {
+	private EntityDomainType resolveEntityReference(final String entityName) {
 		log.debugf( "Attempting to resolve path [%s] as entity reference...", entityName );
 		EntityDomainType reference = null;
 		try {
-			entityName = creationContext.getJpaMetamodel().qualifyImportableName( entityName );
-			reference = creationContext.getJpaMetamodel().entity( entityName );
+			String importName = creationContext.getJpaMetamodel().qualifyImportableName( entityName );
+			if ( entityName.equals( importName ) && getCreationOptions().useStrictJpaCompliance() ) {
+				throw new StrictJpaComplianceViolation(
+						"Encountered FQN entity name [" + entityName + "], but strict JPQL compliance was requested",
+						StrictJpaComplianceViolation.Type.FQN_ENTITY_NAME
+				);
+			}
+			reference = creationContext.getJpaMetamodel().entity( importName );
```

Comment:
The check here should be based on `reference.getName()` like I said on the other one.  

Why?  Well conceptually it is more correct.  And practically it would allow the exception message to include the entity name they should use in "compliance mode".  Right now, your error message does not tell them the corrective action.

Also, `getCreationOptions().useStrictJpaCompliance()` is a more performant check compared to the String comparison - it should come first.